giles-eco-tardis/src/main/java/edu/asu/diging/tardis/core/service/impl/ImageExtractionManager.java

```diff
@@ -73,110 +90,86 @@ public void init() {
-
-                try {
-                    PDPage page = pdfDocument.getPage(i);
-                    
-                    // saves images in page to filesystem
-                    processor.processPage(page);
-                    
-                    
-                    for (String filename : processor.getImageFilenames()) {
-                        PageElement pageElem = new PageElement();
-                        pageElem.setContentType("image/png");
-                        pageElem.setFilename(filename);
-                        pageElem.setType("IMAGE");
-                        pageElem.setDownloadUrl(
-                                restEndpoint + DownloadFileController.GET_FILE_URL
-                                        .replace(
-                                                DownloadFileController.REQUEST_ID_PLACEHOLDER,
-                                                request.getRequestId())
-                                        .replace(
-                                                DownloadFileController.DOCUMENT_ID_PLACEHOLDER,
-                                                request.getDocumentId())
-                                        .replace(DownloadFileController.FILENAME_PLACEHOLDER,
-                                                filename));
-                        pageElem.setStatus(PageStatus.COMPLETE);
-                        requestPage.getPageElements().add(pageElem);
-                    }
-                    
-                } catch (IOException | RuntimeException e) {
-                    messageHandler.handleMessage("Could not render image.", e, MessageType.ERROR);
-                    requestPage.setStatus(PageStatus.FAILED);
-                    requestPage.setErrorMsg(e.getMessage());
-                }
-
-                // this needs to be reset for every page
-                processor.resetFilenames();
-                pages.add(requestPage);
-            }
-            
-            try {
-                pdfDocument.close();
-            } catch (IOException e) {
-                messageHandler.handleMessage("Error closing document.", e, MessageType.ERROR);
-            }
-        }
-
-        progressManager.setPhase(ProgressPhase.WIND_DOWN);
-        ICompletionNotificationRequest completedRequest = null;
+        String imagePath;
         try {
-            completedRequest = requestFactory.createRequest(request.getRequestId(), request.getUploadId());
-        } catch (InstantiationException | IllegalAccessException e) {
-            messageHandler.handleMessage("Could not create request.", e, MessageType.ERROR);
-            // this should never happen if used correctly
+           imagePath = imageFileStorageManager.saveImageFile(imageFile, request);
+        } catch (IOException e) {
+            messageHandler.handleMessage("Could not save image file " + request.getDownloadPath(), e, MessageType.ERROR);
+            return;
```

Comment:
In the previous case when an error is thrown, the status is set to failed but the method doesn't return (presumable to send a failed request completion message). Here it returns. This should be consistent. We probably want to send a failed status to indicate that a component couldn't complete its work. Also, if imageFile is null, then line 112 should not be called.